CONTRIBUTING.md

```diff
@@ -132,9 +132,12 @@ versioning guidelines:
   for further details.
 * When all of the tests are passing and all other conditions described herein are satisfied, a
   maintainer will be assigned to review and merge the PR.
-* Once you submit a PR, *please do not rebase it*. It's much easier to review if subsequent commits
-  are new commits and/or merges. We squash rebase the final merged commit so the number of commits
-  you have in the PR don't matter.
+* Once you submit a PR, *please do not rebase it*. If you rebase, you will need to force push to
+  github, and github's user interface will force your reviewer to review the PR
+  from stratch rather than simply look at your latest changes.  It's much easier for folks to review
```

Comment:
suggest:
  s/for folks to review/to review/

I feel like people don't necessarily know an alternative flow to use if they can't force-push. My naive one is:
```
```
branch=$(git status|head -1|cut -f3 -d\ )
git checkout main
git fetch upstream
git reset --hard upstream/main
git push origin main --force --no-verify
git checkout "$branch"
git merge main
```

should we offer this?

Reply:
fwiw I do not remember how I came up with this sequence, but it's in my "git-sync" script.

Reply:
can we make an exception for WIP prs please?

mostly i tell people not to rebase/force once there are reviewers

there are also some occasions where i think its necessary - not sure exactly why, but github loses the base branch and sees lots of commits as part of the pr, in this case i think its necessary to rebase/force, but there may be another way to resolve

Reply:
I agree that WIP PRs with no reviews so far don't need the restriction, though with the above script I've never needed to force-push.

Reply:
the other situation where i force even after reviewers...

if i add commit, and know the reviewer wont review for some time, and then want to add another commit, sometimes i will squash the two and force - that way the existing review isnt broken, but there are less additional commits to review

Reply:
>  github loses the base branch

it may be the local `main` being out of sync, but trying to rem my own experiences of it i dont think so - not sure why it occurs, its ~frequent

Reply:
agree once reviewers are assigned.  I totally force push when I insta-fail format and there's no reviewer. :-) Updated